nixos/modules/services/misc/jellyfin.nix

```diff
@@ -0,0 +1,59 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.jellyfin;
+in
+{
+  options = {
+    services.jellyfin = {
+      enable = mkEnableOption "Jellyfin Media Server";
+
+      user = mkOption {
+        type = types.str;
+        default = "jellyfin";
+        description = "User account under which Jellyfin runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "jellyfin";
+        description = "Group under which jellyfin runs.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.jellyfin = {
+      description = "Jellyfin Media Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = rec {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        StateDirectory = "jellyfin";
+        CacheDirectory = "jellyfin";
+        ExecStart = "${pkgs.jellyfin}/bin/jellyfin --datadir '/var/lib/${StateDirectory}' --cachedir '/var/cache/${CacheDirectory}'";
```

Comment:
Why does `emby` need to call `chmod` on the plugins folder in the `preStart` script but `jellyfish` does not?

Reply:
From quick glance at the Jellyfin source file that caused #46959, it doesn't seem to me that Jellyfin copies plugins from the nix store to the data dir, so we shouldn't have that problem anymore. But I'll have to do more testing before I'm sure of that

Reply:
Should jellyfish do the same thing as emby? I only ask because it is a relatively young fork so I assume it hasn't diverged too far from emby yet?

Reply:
@nyanloutre I believe your version of this service copied plugins in the `preStart` - can you comment?

Reply:
Ok, so I have finallly some time to test some stuff. As stated in the issue linked above, Emby copies plugins from the nix store, but Jellyfin doesn't have a plugins directory in its installation anymore. It seems that they have moved all "builtin" plugins to a plugin store, and download them dynamically (so no permission issues). In fact, just to be sure, a `find` of the name `plugins` in jellyfin's installation directory just returns the HTML and JS files of the plugins page.

Reply:
If the plugins are distributed as binaries will they run under NixOS? :anguished:

Reply:
I think there should be no issue, plugins are distributed as DLL, that are loaded by the already "up and running with all the right flags" dotnet. I don't know much about C#, but I tested the Open Subtitles plugin and everything seems good ^^